### PR TITLE
Combined branch

### LINKED ISSUE #92: Reachability request kan sendes fra current state
uden clock values

### LINKED ISSUE #97: Reachability: Show "not reachable" response

### LINKED ISSUE #99: Reachability: Show response on components
Få edgeid's ud af response og highlight dem

### LINKED ISSUE #101: SIM: test

### LINKED ISSUE #102: SIM: Clock values i trace

### LINKED ISSUE #103: SIM: Clock values når man klikker på top right corner

### LINKED ISSUE #107: Refactor static properties
- [x] ListOfComponents => ComponentsInSimulation, placed in SimulationHandler as non-static
- [x] SimulationQuery => placed in SimulationHandler and non-static
- [ ] Fix tests

### LINKED ISSUE #123: Ret highlight i reachability så det adskilles fra simulation

### LINKED ISSUE #128: fjern rester af reachability på location (ikke sim)
Der ligger en del "rod", fx UpdateReachability i LocationPresentation (vi får kaldt getLocationReachableQuery, når vi tilføjer en ny location i editorview. Tænker at det er unødvendigt)

### LINKED ISSUE #135: Ret brug af Ecdar.getSimulationHandler (kun én gang per klasse)

### PR DESCRIPTION
**Pull request by EmilieSonne**
This issue contains final changes to simulation and reachability checks. 
It solves the following issues: fixes #103 , fixes #128 , fixes #123 , fixes #107 , fixes #101 , fixes #102 , fixes #99 , fixes #97 ish ;), fixes #92 and fixes #135.

In plain text, we are now able to make a reachability check from current locations (in the simulation view, where stepping changes the state). Clocks are not taken into consideration as the simulation results are not compatible with the expected input from the reachability part. The trace/path resulting from a successful reachability check is highlighted in a deep purple color. 

Improvements to the simulation view are that it now displays the clock values in the trace, and it is possible to jump between different states without loosing the trace history. It is also possible to get clocks by pressing the small symbol in the top right corner in the simulation view now. 

On a more general note, we have also tried to conquer the use of static/global properties. (den her går ud til Sigurd)

Finally, test cases have been added to the simulation view. 

We wish you good luck. Ses til julefrukost (husk at tilmelde jer please).

### CODE REVIEW

**Review by mksm20 — CHANGES_REQUESTED** [review]
Se kommentarer, ser fornuftigt ud!

File: src/main/java/ecdar/controllers/EcdarController.java [hunk 1]
```diff
@@ -705,7 +625,7 @@ private void initializeViewMenu() {
                     break;
                 }
             }
-            scaling.selectToggle(scaleM); // Necessary to avoid project pane appearing off-screen
+            // scaling.selectToggle(scaleM); // Necessary to avoid project pane appearing off-screen
```
[the remark this answers is not in the document]
Comment (jhbengtsson) [reply src/main/java/ecdar/controllers/EcdarController.java]: upsi, den er væk nu :)

File: src/main/java/ecdar/abstractions/BackendInstance.java [hunk 2]
```diff
@@ -114,10 +114,11 @@ public JsonObject serialize() {
 
     @Override
     public void deserialize(final JsonObject json) {
+        setIsThreadSafe(true);
```
[the remark this answers is not in the document]
Comment (jhbengtsson) [reply src/main/java/ecdar/abstractions/BackendInstance.java]: den er fjernet nu

**Review by mksm20 — APPROVED** [review]
lgtm

**Review by Mightyhaha — APPROVED** [review]
LGTM, Ses til juleFRUkost

**Pull request merged**